api/evidence.go

```diff
@@ -30,6 +31,16 @@ type Evidence struct {
 	UpdatedAt        time.Time     `json:"updated_at"`
 }
 
+func (evidence *Evidence) BeforeCreate(tx *gorm.DB) (err error) {
+	if evidence.CreatedBy == uuid.Nil {
+		evidence.CreatedBy = *SystemUserID
+	}
+	if evidence.ID == uuid.Nil {
+		evidence.ID = uuid.New()
+	}
```

Comment:
Do we need this ? can't we do that via schema ?

Reply:
We can either do it in here or in the gorm struct tag. I guess struct tag is more idiomatic.